stdlib/universe/quantile.go

```diff
@@ -258,32 +252,62 @@ func (a *QuantileAgg) NewUIntAgg() execute.DoUIntAgg {
 }
 
 func (a *QuantileAgg) NewFloatAgg() execute.DoFloatAgg {
-	return a.Recycle()
+	q := &QuantileAggState{
+		parent: a,
+	}
+	if len(a.freeDigests) > 0 {
+		q.digest = a.popFreeDigest()
+	} else {
+		a.mem.Account(tdigest.ByteSizeForCompression(a.Compression))
+		q.digest = tdigest.NewWithCompression(a.Compression)
+	}
+	return q
 }
 
 func (a *QuantileAgg) NewStringAgg() execute.DoStringAgg {
 	return nil
 }
 
-func (a *QuantileAgg) DoFloat(vs *array.Float) {
+func (a *QuantileAgg) Dispose() {
+	for i := 0; i < len(a.freeDigests); i++ {
+		a.mem.Account(tdigest.ByteSizeForCompression(a.Compression) * -1)
+	}
```

Comment:
Can you add an assignment here:
```go
a.freeDigests = nil
```
This way Go's garbage collector will see that the slice is unreachable and will be more likely to free the memory sooner. It might not make much of a difference, but it can't hurt.